OrbitGl/App.cpp

```diff
@@ -272,7 +272,28 @@ void OrbitApp::PostInit() {
 
     auto callback = [&](ProcessManager* process_manager) {
       main_thread_executor_->Schedule([&, process_manager]() {
-        m_ProcessesDataView->SetProcessList(process_manager->GetProcessList());
+        std::vector<ProcessInfo> process_list =
+            process_manager->GetProcessList();
+
+        uint32_t highest_cpu_pid = 0;
+        if (m_ProcessesDataView->GetSelectedProcessId() == 0) {
```

Comment:
Sorry, I don't know how the sorting is processed and how the sorted list is represented, so maybe this is not straightforward to do, but isn't there a more direct way of picking the process that is shown first in the list?

Reply:
Isn't looking for the maximum in the list a direct way? Or what do you mean?

Reply:
I guess the idea was selecting the first ui element in the list as it should be sorted already.  However, I'm not sure that the UI is actually ready at that point in the PostInit.  Are we guaranteed to have a received the list of processes by that point?

Reply:
The list that is displayed is sorted based on which column is selected as the sorting column in the ui. Sure, it's the CPU column by default, but I don't think we should rely on the order in which things are displayed in the ui. Also, in theory the user could change the sorting column before the list is received. And in general, I don't think this code should be related to ui logic.

Reply:
Yes I agree.  Please disregard my question about the guarantee of having received the process list, I overlooked the fact that this code runs as a callback.

Reply:
Yes, the idea was to select the first UI element in the list based on the assumption that the UI is already in the right state (process list received and sorted; is that the case?) and that this would be the smallest code change. All this is only a heuristic for the problem "which process does the user want to profile", so I think it's fine if we get it wrong in edge cases (which we also might with the highest CPU load). 

If the user has changed the sorting order and you select the process with the highest CPU load, there's also the problem that this process might not be visible, but it would still be selected. That might be confusing (but as all other edge cases, would be very rare anyway).

Reply:
Even if there could be an even smaller code change, I think this is small enough. As I said, I don't think this code should rely on ui state/code when it's not necessary.

Reply:
Although to be fair this already relies on `m_ProcessesDataView->GetSelectedProcessId()`. Mmmh.

Reply:
I was hoping we can do this with close to a single line change. But I'm not close enough to the code here to know if that's possible. 

> As I said, I don't think this code should rely on ui state/code when it's not necessary.

I'm not sure I understand this. Isn't the selected process part of the UI state?

Reply:
I created #688, let me know if you prefer it.